frontend/src/components/common/ApiVersionInfo.tsx

```diff
@@ -11,11 +11,17 @@ export const ApiVersionInfo = () => {
   const [version, setVersion] = React.useState<IApiVersion>();
 
   React.useEffect(() => {
+    let isActive = true;
     const get = async () => {
       const response = await getVersion();
-      setVersion(response.data);
+      if (isActive) {
+        setVersion(response.data);
```

Comment:
only set the version if this component is active, fixes a console warning.

Reply:
I don't understand the issue here?

Reply:
@Fosol maybe the full stack trace will help understand

> Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in ApiVersionInfo (at Footer.tsx:16)
    in div (created by styled.div)
    in styled.div (at Footer.tsx:11)
    in Footer (at EmptyLayout.tsx:24)
    in div (created by Container)
    in Container (at EmptyLayout.tsx:23)
    in footer (created by styled.footer)
    in styled.footer (at Footer.tsx:12)
    in Footer (at EmptyLayout.tsx:22)
    in div (created by Container)
    in Container (at EmptyLayout.tsx:13)
    in EmptyLayout (at src/index.tsx:40)

Reply:
Basically the `useEffect` is running while the ApiVersionInfo component has not rendered yet so react complains about it. Devin's code fixes that

Reply:
This is a common issue with useEffects that have async callbacks. If the callback is being executed when the component is being cleaned up, you will see a console warning as state changes are not allowed on inactive components.

This check uses a closure to ensure that the state is only set when the component is active.

Here is an article describing the issue: https://juliangaramendy.dev/blog/use-promise-subscription